### PR TITLE
CI: Add linter check for scala

### PR DESCRIPTION
**Pull request by ljubon**
The goal of this PR is to enable CI to have linter enabled and basic rules defined in `.github/workflows/linter.yml`

### CODE REVIEW

File: .scalafmt.conf [hunk 1]
```diff
@@ -0,0 +1,6 @@
+version = 3.7.14
+
+runner.dialect = scala213
+
```
Comment (EnricoMi) [inline .scalafmt.conf:4]:
```suggestion
+rewrite.trailingCommas.style = keep

```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -23,15 +23,21 @@ jobs:
           name: Event File
           path: ${{ github.event_path }}
 
+  lint:
+    name: "Run linter"
```
Comment (EnricoMi) [inline .github/workflows/ci.yml:27]:
```suggestion
    name: "Linting"
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -23,15 +23,21 @@ jobs:
...
+    needs: lint
     uses: "./.github/workflows/build-jvm.yml"
   build-snapshots:
     name: "Build Snapshots"
+    needs: lint
     uses: "./.github/workflows/build-snapshots.yml"
   build-python:
     name: "Build Python"
-    needs: build-jvm
+    needs: [build-jvm,lint]
```
Comment (EnricoMi) [inline .github/workflows/ci.yml:40]:
```suggestion
    needs: [build-jvm, lint]
```

**Review by EnricoMi — CHANGES_REQUESTED** [review]

**Review by EnricoMi — CHANGES_REQUESTED** [review]

**Comment by EnricoMi** [comment]
How can we run scalafmt from command line / mvn so that developer can auto-format the code?

**Comment by EnricoMi** [comment]
We can make scalafmt part of `mvn compile` so that this is checked during development:
- add to `pom.xml`: https://github.com/G-Research/spark-extension/pull/201/files#diff-9c5fb3d1b7e3b0f54bc5c4182965c4fe1f9023d449017cece3005d3f90e8e4d8
- run in compile phase: https://github.com/G-Research/spark-extension/pull/212/files#diff-9c5fb3d1b7e3b0f54bc5c4182965c4fe1f9023d449017cece3005d3f90e8e4d8R247-R256

No need to add this specifically to CI then.

**Comment by pavlovic-ivan** [comment]
I am good with suggested as well

**Comment by EnricoMi** [comment]
All merged.

**Pull request closed without merging**